### PR TITLE
Derive Eq for sorted maps via data equality

### LINKED ISSUE #491: Make `PMap` equality use data equality

### PR DESCRIPTION
**Pull request by MangoIV**
- closes #491 
- forgot to make clean branch and hence need #511

### CODE REVIEW

File: Plutarch/Api/V1/AssocMap.hs [hunk 1]
```diff
@@ -63,14 +64,24 @@ import Plutarch.Unsafe (punsafeCoerce, punsafeDowncast)
...
 data KeyGuarantees = Sorted | Unsorted
 
 type role PMap nominal nominal nominal nominal
 newtype PMap (keysort :: KeyGuarantees) (k :: PType) (v :: PType) (s :: S) = PMap (Term s (PBuiltinMap k v))
   deriving stock (Generic)
-  deriving anyclass (PlutusType, PIsData, PEq, PShow)
+  deriving anyclass (PlutusType, PIsData, PShow)
 instance DerivePlutusType (PMap keysort k v) where type DPTStrat _ = PlutusTypeNewtype
 
+instance PEq (PMap 'Unsorted k v)
```
Comment (L-as) [inline Plutarch/Api/V1/AssocMap.hs:77]: I don't think this is correct. This instance should be removed, no?

Comment (MangoIV) [reply Plutarch/Api/V1/AssocMap.hs:77]: is that not how it was derived before for unsorted map? this is the same as anyclass deriving

Comment (L-as) [reply Plutarch/Api/V1/AssocMap.hs:77]: yep, and I'm pretty sure this is incorrect

Comment (MangoIV) [reply Plutarch/Api/V1/AssocMap.hs:77]: What do you mean? Like in you don't want two maps that are equal modulo sortedness to be seen as unequal? I think that's fine tbh.

Comment (MangoIV) [reply Plutarch/Api/V1/AssocMap.hs:77]: So you just want me to remove it? that means we don't have any notion of equality for Unsorted Maps?

Comment (L-as) [reply Plutarch/Api/V1/AssocMap.hs:77]: Yes, it's confusing.

Comment (L-as) [reply Plutarch/Api/V1/AssocMap.hs:77]: You can always just pto then compare.

Comment (MangoIV) [reply Plutarch/Api/V1/AssocMap.hs:77]: okay then, lemme remove it

**Review by L-as — APPROVED** [review]
Great!

**Pull request merged**